src/sources/exec/mod.rs

```diff
@@ -394,11 +405,30 @@ async fn run_command(
                 &mut event,
             );
 
-            if out.send(event).await.is_err() {
-                error!(message = "Failed to forward event; downstream is closed.");
-                break 'send;
+            match out.send(event).await {
+                Ok(_) => {
+                    processed_size += event_size;
+                    processed_count += 1;
+                }
+                Err(error) => {
+                    // we need to emit the successfully sent events before breaking the while loop
+                    emit!(&ExecEventsSent {
+                        count: processed_count,
+                        byte_size: processed_size,
+                    });
+                    emit!(&ExecStreamError {
+                        count: total_count - processed_count,
+                        byte_size: total_size - processed_size,
+                        error: error.to_string(),
+                    });
```

Comment:
If we don't have a common error for this, we should create one. It is a circumstance generic to all sources.

Reply:
done with `StreamClosedError` 👍